soh/soh/Enhancements/debugger/actorViewer.cpp

```diff
@@ -0,0 +1,1261 @@
+#include "actorViewer.h"
+#include "../../util.h"
+#include "../libultraship/SohImGuiImpl.h"
+#include "ImGuiHelpers.h"
+
+#include <array>
+#include <bit>
+#include <map>
+#include <string>
+
+extern "C" {
+#include <z64.h>
+#include "z64math.h"
+#include "variables.h"
+#include "functions.h"
+#include "macros.h"
+extern GlobalContext* gGlobalCtx;
+
+#include "textures/icon_item_static/icon_item_static.h"
+#include "textures/icon_item_24_static/icon_item_24_static.h"
+}
+
+typedef struct {
+    u16 id;
+    u16 params;
+    Vec3f pos;
+    Vec3s rot;
+} ActorInfo;
+
+typedef enum {
+    LIST,
+    TARGET,
+    HELD,
+    INTERACT
+} RetrievalMethod;
+
+std::vector<std::string> acMapping = { 
+    "Switch",
+    "Background (Prop type 1)",
+    "Player",
+    "Bomb",
+    "NPC",
+    "Enemy",
+    "Prop type 2",
+    "Item/Action",
+    "Misc.",
+    "Boss",
+    "Door",
+    "Chest"
+};
+
+// https://wiki.cloudmodding.com/oot/Actor_List_(Variables)
+const std::string GetActorDescription(u16 id) {
```

Comment:
I'd say this makes more sense as a `std::map`  and we can't do anything with the unset actors so they can just be removed.